### ISSUE TITLE
Fix unstable error calculation in activations with vanishing gradients (Mish, Swish, GELU, etc.)

### ISSUE THREAD

**Issue by nleck** [stSoftwareAU/NEAT-TS#1588]
## Problem
Several activation functions (`Mish`, `Swish`, `GELU`, `ELU`, `Softplus`, etc.) calculate backpropagation error using `error = rawError / derivative`.
In regions where the derivative is near zero (e.g., large negative inputs for Mish/Swish), this division results in extremely large error values (often hitting the clamp limit of 100).

## Impact
This causes 'exploding gradients' in the error signal, destabilizing training, especially for complex or deep networks where neurons might be in saturation. This aligns with reports of training failing for complex creatures.

## Affected Files
- `src/methods/activations/types/Mish.ts`
- `src/methods/activations/types/Swish.ts`
- `src/methods/activations/types/GELU.ts`
- `src/methods/activations/types/ELU.ts`
- `src/methods/activations/types/SELU.ts`
- `src/methods/activations/types/Softplus.ts`
- `src/methods/activations/types/LogSigmoid.ts`
- ... and potentially others using `rawError / slope` without a fallback.

## Proposed Solution
Refactor `calculateError` in these classes to:
1.  Check if the derivative (slope) is too small.
2.  If small, fall back to `unSquash` (inverse function) to find the target input value and calculate `targetInput - currentInput`.
3.  If `unSquash` is expensive or unavailable, implement a robust approximation or handle the 'dead zone' gracefully (like `ReLU` does).

## Reproduction
Test case:
```typescript
const activation = new Mish();
const input = -10; // Derivative is ~0
const target = 0; // Current activation is ~0
// Expectation: Error should be small (adjust input slightly)
// Actual: Error is clamped to 100 because derivative is tiny.
```
